### DISCUSSION TITLE
Use the Router decorator to add custom routes

### DISCUSSION THREAD

**Discussion by Wingrammer** [adrien2p/medusa-extender#99] [Q&A] [answered]
Hello,

I am wondering if there is a special requirement for the controler to be passed in the handlers array in order to create a custom route.

`import { Router } from 'medusa-extender';
import createUserHandler from '@medusajs/medusa/dist/api/routes/admin/users/create-user';
import wrapHandler from '@medusajs/medusa/dist/api/middlewares/await-middleware';
import {Request, Response} from 'express

@Router({
    routes: [
        {
            requiredAuth: false,
            path: '/admin/test',
            method: 'get',
            handlers: [(req:Request, res:Response) => res.status(200)],
        }
    ],
})
export class UserRouter {
}`

Having tried with the code above the code compiled without error. But the requests am sending never end.
I have tried passing my controller to wrapHandler as well but just got an error: fn.apply(...).catch is not a function.

What am I missing to successfully add and use a custom route using medusa-extender Router decorator?

**Comment by adrien2p** [comment, marked as answer]
Hi there, I suggest that those kind of questions are created under an issue and not a discussion for the next time. So that we keep the discussion for discussion purpose ☺️

First of all, your handler never send any response, it is just setting a status and that's it.

For more example, you can look at the integration tests that are a good source of examples here https://github.com/adrien2p/medusa-extender/tree/main/integration-tests/tests
